bitbots_ball_filter:
  ros__parameters:    
    # Ball parameters:
    # Topics
    ball_subscribe_topic: 'balls_relative'
    ball_position_publish_topic: 'ball_position_relative_filtered'
    ball_movement_publish_topic: 'ball_relative_movement'
    ball_publish_topic: 'ball_relative_filtered'

    ball_filter_reset_service_name: 'ball_filter_reset'

    filter_frame: 'odom'
    #filter_frame: 'map'

    # Filtering rate in Hz
    filter_rate: 62  # we get an image every 16 ms -> 62.5 hz

    # Velocity reduction (per axis) factor of the ball per second
    velocity_reduction: 0.6

    # Noise which is added to the estimated position without new measurements
    process_noise_variance: 0.001

    # Ball measurement certainty in filter
    measurement_certainty: 0.1

    # Max ball not seen time in Seconds
    filter_reset_time: 20

    # Distance to the current estimation causing a filter reset
    filter_reset_distance: 2

    # True if ball should be selected based on distance to filtered ball instead of highest rating
    closest_distance_match: True

    # Robot parameters:
    # Topics
    robot_subscribe_topic: 'robots_relative'
    robot_position_publish_topic: 'robot_position_relative_filtered'
    robot_movement_publish_topic: 'robot_relative_movement'
    robot_publish_topic: 'robot_relative_filtered'

    robot_filter_reset_service_name: 'robot_filter_reset'

    robot_filter_frame: 'odom'
    #filter_frame: 'map'

    # Filtering rate in Hz
    robot_filter_rate: 62  # we get an image every 16 ms -> 62.5 hz

    # Velocity reduction (per axis) factor of the robot per second
    robot_velocity_reduction_x: 9.59 #0.6
    robot_velocity_reduction_y: 5.36 #0.6

    # Noise which is added to the estimated position without new measurements
    robot_process_noise_variance: 0.70 #0.001

    # Robot measurement certainty in filter
    robot_measurement_certainty: 0.67 #0.1

    # Max robot not seen time in Seconds
    robot_filter_reset_time: 68 #20

    # Distance to the current estimation causing a filter reset
    robot_filter_reset_distance: 7 #2

    # True if robot should be selected based on distance to filtered robot instead of highest rating
    robot_closest_distance_match: True #True

    # todo are these really x and y?
    robot_transition_modifier_x: 0.12 #1.0
    robot_transition_modifier_y: 2.35 #1.0

    # Optimizer parameters:
    # todo this doesn't work yet
    max_filter_cycles: 100
    optuna_trials: 10

    # string containing all parameters from Optuna
    adjusted_params: ""
    trial_number: 0
    selfdestruct: False
